tests/scale_test/notifier.py

```diff
@@ -0,0 +1,239 @@
+
+    # Send notifications
+    start = timeit.default_timer()
+    response = utils.send_notification(lira_url, auth_token, notification)
+    stop = timeit.default_timer()
+
+    # Output metrics
+    logging.info('Sent notification with Round-Trip-Time: {rtt}, in {total_time} seconds, status is {status}'.format(
+        rtt=response.elapsed.total_seconds(),
+        total_time=stop - start,
+        status=response.status_code
+    ))
+
+
+@notify.command(short_help='Send notifications from a file of a list of bundle UUIDs and VERSIONs.')
+@click.pass_context
+@click.option('--file',
+              help='The path to the file of a list of bundles',
+              type=click.Path(exists=True))
+@click.option('--async',
+              'run_mode',
+              flag_value='async',
+              default=True,
+              help='Asynchronously send notifications, faster.',
+              show_default=True)
+@click.option('--sync',
+              'run_mode',
+              flag_value='sync',
+              help='Synchronously send notifications, linear and slower.',
+              show_default=True)
+def batch(ctx, file, run_mode):
+
+    # Using a custom checker instead of the required=True flag of Click, so it shows the help text
+    utils.required_checker(ctx, file=file)
+
+    # Checking if the auth_token given is correct
+    auth_token = utils.auth_checker(ctx)
+
+    # Prepare arguments
+    bundles = utils.read_bundles(file)
+    lira_url, label, save_path = ctx.obj['lira_url'], ctx.obj['label'], ctx.obj['save_path']
+
+    if run_mode == 'async':
+        logging.info('Sending notifications asynchronously...\n')
+        async_notify(bundles, lira_url, label, auth_token, save_path)
+    elif run_mode == 'sync':
+        logging.info('Sending notifications synchronously...\n')
+        linear_notify(bundles, lira_url, label, auth_token, save_path)
+
+
+def linear_notify(bundles, lira_url, label, auth_token, save_path):
+    # Use a probe to get the current subscription_id
+    subscription_id = utils.subscription_probe(lira_url)
+    logging.info('Using subscription_id: {}'.format(subscription_id))
+
+    # Prepare the payload
+    queue = deque([])
+    for bundle in bundles:
+        queue.append(utils.prepare_notification(bundle['bundle_uuid'], bundle['bundle_version'], subscription_id, label))
+
+    # Send notifications
+    start = timeit.default_timer()
+    responses = [utils.send_notification(lira_url, auth_token, notification) for notification in tqdm(queue)]
+    stop = timeit.default_timer()
+
+    # Save metrics
+    rtt = [response.elapsed.total_seconds() for response in responses]
+    status = [response.status_code for response in responses]
+    total_time = stop-start
+    save_file = pathlib.Path(save_path) / 'notifier_metrics_{}.json'.format(time.strftime("%Y%m%d-%H%M%S"))
+    utils.dump_metrics(save_file, rtt=rtt, status=status, total_time=total_time)
```

Comment:
Can you also add the log statement `logging.info('Saved the metrics file to {}'.format(save_file))` here? You could also consider moving this block of code into a separate function, since it's the same in both the `linear_notify` and `async_notify` functions

Reply:
Nice catch!